crowbar_framework/app/models/api/storage.rb

```diff
@@ -15,6 +15,27 @@
 #
 
 module Api
-  class Storages
+  class Storage < Tableless
+    def repocheck
+      {}.tap do |ret|
+        ret["os"] = {}
+        ret["ceph"] = {}
+
+        # get target platform
+        # FIXME: get rid of platform_exception when SES upgraded to SP2
+        platform = Api::Upgrade.new.target_platform(platform_exception: :ceph)
+
+        ["os", "ceph"].each do |feature|
+          unless ::Crowbar::Repository.provided_and_enabled?(feature, platform)
+            Openstack::Upgrade.enable_repos_for_feature(feature, Rails.logger)
+          end
+          available, repolist = ::Crowbar::Repository.provided_and_enabled_with_repolist(
+            feature, platform
+          )
+          ret[feature]["available"] = available
+          ret[feature]["repos"] = repolist.deep_stringify_keys
```

Comment:
So I would still simplify the return value. To me, `ret` should only include a `ceph` attribute. So that would be something like:

```ruby
        ret["ceph"] = {}

        platform = Api::Upgrade.new.target_platform(platform_exception: :ceph)

        ["os", "ceph"].each do |feature|
          unless ::Crowbar::Repository.provided_and_enabled?(feature, platform)
            Openstack::Upgrade.enable_repos_for_feature(feature, Rails.logger)
          end
          available, repolist = ::Crowbar::Repository.provided_and_enabled_with_repolist(
            feature, platform
          )
          ret["ceph"]["available"] &&= available
          ret["ceph"]["repos"].deep_merge(repolist.deep_stringify_keys)
        end
```

(the `deep_merge` bit is probably not correct, but hopefully it gives the idea)

The point here is that when we check for ceph, the return value that is expected is only for ceph, and yes, it might be that for ceph, the OS repos are missing.

Does it make sense?